app/views/groups/show.html.erb

```diff
@@ -1,5 +1,29 @@
 <h1> This is a test group#show page </h1>
 <br />
-<h1>The group name is <%= @group.name %></h1>
+<h1>The group name is <%= current_group.name %></h1>
 <br />
-<h2><%= @group.description %></h2>
\ No newline at end of file
+<h2><%= current_group.description %></h2>
+
+<% if !current_group.users.include? current_user %>
+
+  <%= link_to 'Join Group',join_group_path(current_group), method: :patch, class: 'btn btn-primary' %>
+
+<% elsif current_group.users.include? current_user %>
+
+  <%= link_to 'Leave Group', leave_group_path(current_group), method: :patch, class: 'btn btn-primary' %>
+
+<% end %>
+
+
+<h1> The users are: </h1>
+<% current_group.users.each do |u| %>
+  <%= u.name %>
+  <br />
+<% end %>

```

Comment:
This works, but best practice is to be verbose with variable names. As this gets a lot larger, it is easier to see what a variable named `user` is referencing vs. a variable named `u`.